src/filter_tree/filter_tree.c

```diff
@@ -612,9 +628,76 @@ static bool _FilterTree_Compact_Or(FT_FilterNode *node) {
 	}
 }
 
+// Compacts 'XOR' and 'XNOR' condition nodes.
+static bool _FilterTree_Compact_XOr(FT_FilterNode *node, bool xnor) {
```

Comment:
Consider moving all `FilterTree_Compact*` logic into a new file `filter_tree_compact.c`

Reply:
we can't do it because _FilterTree_Compact_Pred uses _applyFilter which is static and can't be moved